### PR TITLE
supporting append blobs in blob triggers

### PR DESCRIPTION
**Pull request by brettsam**
Addresses #858.

### CODE REVIEW

File: src/Microsoft.Azure.WebJobs.Storage/Blob/IStorageAppendBlob.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+#if PUBLICSTORAGE
+namespace Microsoft.Azure.WebJobs.Storage.Blob
+#else
+namespace Microsoft.Azure.WebJobs.Host.Storage.Blob
+#endif
+{
+    /// <summary>Defines a block blob.</summary>
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Storage/Blob/IStorageAppendBlob.cs]: nit: fix comment

File: src/Microsoft.Azure.WebJobs.Storage/Blob/StorageAppendBlob.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+#if PUBLICSTORAGE
+namespace Microsoft.Azure.WebJobs.Storage.Blob
+#else
+namespace Microsoft.Azure.WebJobs.Host.Storage.Blob
+#endif
+{
+    /// <summary>Represents a page blob.</summary>
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Storage/Blob/StorageAppendBlob.cs]: nit: fix comment

**Comment by mathewc** [comment]
Have you done some e2e Dashboard testing with this? For the various blob bindings (e.g. CloudBlobStreamValueBinder) they implement IWatchable, allowing the Dashboard to show in progress updates for parameters, in the case of long running functions that upload a bunch of data over time to a blob. I want to make sure that all works for append blobs as well.

**Comment by brettsam** [comment]
Also now supports #859.

**Comment by mathewc** [comment]
What's the status of this one?

File: src/Microsoft.Azure.WebJobs.Storage/Blob/IStorageAppendBlob.cs (hunk 1 above)
Comment (brettsam) [inline src/Microsoft.Azure.WebJobs.Storage/Blob/IStorageAppendBlob.cs]: Fixed.

**Comment by brettsam** [comment]
Waiting for final signoff -- the latest dashboard bits all looked good. Note that I also added support for `CloudAppendBlob` input bindings (#859) with my last commit b/c it was pretty easy to get working.

**Comment by mathewc** [comment]
:ship:

**Pull request merged**